lib/apply-extends.js

```diff
@@ -0,0 +1,41 @@
+var fs = require('fs')
+var path = require('path')
+var assign = require('./assign')
+var YError = require('./yerror')
+
+var previouslyVisitedConfigs = []
+
+function checkForCircularExtends (path) {
+  if (previouslyVisitedConfigs.indexOf(path) > -1) {
+    throw new YError("Circular extended configurations: '" + path + "'.")
+  }
+}
+
+function getPathToDefaultConfig (cwd, pathToExtend) {
+  return path.isAbsolute(pathToExtend) ? pathToExtend : path.join(cwd, pathToExtend)
```

Comment:
Yes, I believe `path.resolve()` would work here. I actually thought that `path.join()` worked the same way, which is why I said "relative or absolute" in the docs, but apparently it doesn't.